### PR TITLE
TC01

### PR DESCRIPTION
**Pull request by viethuynh140499**
update TC01

### CODE REVIEW

File: src/test/java/test_cases/BaseTest.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import common.helpers.BrowserHelper;
+import common.helpers.Constant;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+
+public class BaseTest {
+
+  @BeforeMethod
+  public void beforeMethod() {
+    BrowserHelper.startBrowser(BrowserHelper.DriverType.CHROME);
```
Comment (LoliSlayerMC) [inline src/test/java/test_cases/BaseTest.java:12]: hardcode Chrome here. How do you use Firefox or Edge?

File: src/test/java/test_cases/TestCase01.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  public void test() {
+    User user = new User(Constant.USERNAME, Constant.PASSWORD);
+
+    generalPage.clickLoginTab();
+    loginPage.login(user);
+
+    String actualResult = generalPage.getWelcomeMessage();
+    String expectedResult = "Welcome " + Constant.USERNAME;
+
+    Assert.assertEquals(actualResult, expectedResult, "Welcome" + Constant.USERNAME + " message is not displayed as expected");
```
Comment (LoliSlayerMC) [inline src/test/java/test_cases/TestCase01.java:24]: Repeated code here. Use `expectedResult ` instead of `"Welcome" + Constant.USERNAME`

**Review by LoliSlayerMC — CHANGES_REQUESTED** [review]

File: src/test/java/test_cases/BaseTest.java (hunk 1 above)
Comment (viethuynh140499) [reply src/test/java/test_cases/BaseTest.java:12]: > hardcode Chrome here. How do you use Firefox or Edge?

if using FireFox or Edge, I change CHROME to FIREFOX or EDGE. All 3 I have declared the in the BrowserHelper class and  in the startBrowser method

- FireFox 
`BrowserHelper.startBrowser(BrowserHelper.DriverType.FIREFOX);`

-Edge
`BrowserHelper.startBrowser(BrowserHelper.DriverType.EDGE);`

**Comment by viethuynh140499** [comment]
Thank you for reviewing my code. I have fixed all the flaws that you pointed out for me, except the second one. The welcome message displays on all pages so I think it is fine to put it in BasePage. In case I get it wrong, hope you will explain it to me.

File: src/main/java/common/helpers/BrowserHelper.java [hunk 3]
```diff
@@ -14,6 +14,8 @@
 
 public class BrowserHelper {
 
+    private static final BrowserHelper instance = new BrowserHelper();
```
Comment (LoliSlayerMC) [inline src/main/java/common/helpers/BrowserHelper.java:17]: No need to create new instance. This is a helper class, so make all the necessary methods to public static modifier.

File: src/main/java/common/helpers/BrowserHelper.java [hunk 4]
```diff
@@ -22,6 +24,10 @@ public static void navigateToUrl(String url) {
         getDriver().get(url);
     }
 
+    public static BrowserHelper getInstance() {
+        return instance;
+    }
+
```
Comment (LoliSlayerMC) [inline src/main/java/common/helpers/BrowserHelper.java:30]: Also remove this

**Review by LoliSlayerMC — CHANGES_REQUESTED** [review]

**Comment by LoliSlayerMC** [comment]
> Thank you for reviewing my code. I have fixed all the flaws that you pointed out for me, except the second one. The welcome message displays on all pages so I think it is fine to put it in BasePage. In case I get it wrong, hope you will explain it to me.

Copied pasted à em?

**Comment by viethuynh140499** [comment]
Thank you for review my code. I removed getInstance() method and edit  "instance" into public

**Review by LoliSlayerMC — APPROVED** [review]

**Review by LoliSlayerMC — APPROVED** [review]
LGTM

**Pull request merged**